level2/autoscalinggroup.tf

```diff
@@ -0,0 +1,27 @@
+# Autoscaling group using the main-launch-temp template
+resource "aws_autoscaling_group" "main-autoscaling-group" {
+  name                      = "main-autoscaling-group"
+  max_size                  = 5
+  min_size                  = 2
+  health_check_grace_period = 200
+  health_check_type         = "ELB"
```

Comment:
I'd rather use "EC2"
Otherwise, it will terminate your instances if ELB health check is failed.

You can read more about this property here:
https://docs.aws.amazon.com/autoscaling/ec2/userguide/ec2-auto-scaling-health-checks.html